pymc/distributions/continuous.py

```diff
@@ -1051,31 +925,7 @@ def get_mu_lam_phi(
             "mu and lam, mu and phi, or lam and phi."
         )
 
-    def logp(
-        value,
-        mu: Union[float, np.ndarray, TensorVariable],
-        lam: Union[float, np.ndarray, TensorVariable],
-        alpha: Union[float, np.ndarray, TensorVariable],
-    ) -> RandomVariable:
-        """
-        Calculate log-probability of Wald distribution at specified value.
-
-        Parameters
-        ----------
-        value : tensor_like of float
-            Value(s) for which log-probability is calculated. If the log probabilities for multiple
-            values are desired the values must be provided in a numpy array or Aesara tensor.
-        mu : tensor_like of float
-            Mean of the distribution (mu > 0).
-        lam : tensor_like of float
```

Comment:
This is a really good example. I wouldnt know what lam or alpha are by memory and from the logp calculation below theres not enough math to tell me. Having even just a basic description of the parameters would be helpful mathematically.

The other is the what is allowed as inputs. If I was to write code that called this logp I would know what to pass, and auto type checking would be useful.

In the scope of things I care way more about 1 than I do 2, though both would be nice

Reply:
This is a really good example of why I don't like it. The return type is completely wrong, should be TensorVariable (Type) not RandomVariable (Op). Also we only test and promise to support TensorVariable inputs, not float or np.ndarray (even though many times they will work)

Reply:
The parameter information is contained in the main distribution docstring already.

Reply:
https://github.com/pymc-devs/pymc/blob/d34b557b5e0a0806842e2d4ec5aa5a842dbf8dfe/pymc/distributions/continuous.py#L861-L868